olp-cpp-sdk-dataservice-read/src/StreamLayerClientImpl.cpp

```diff
@@ -20,24 +20,143 @@
 #include "StreamLayerClientImpl.h"
 
 #include <olp/core/cache/DefaultCache.h>
+#include <olp/core/client/CancellationContext.h>
 #include <olp/core/client/OlpClientSettingsFactory.h>
+#include <olp/core/logging/Log.h>
+#include <olp/core/porting/make_unique.h>
+#include <olp/core/thread/TaskScheduler.h>
+#include "ApiClientLookup.h"
+#include "Common.h"
+#include "generated/api/StreamApi.h"
+#include "repositories/ExecuteOrSchedule.inl"
 
 namespace olp {
 namespace dataservice {
 namespace read {
 
+namespace {
+constexpr auto kLogTag = "StreamLayerClientImpl";
+constexpr auto kSubscriptionModeParallel = "parallel";
+constexpr auto kSubscriptionModeSerial = "serial";
+constexpr auto kStreamService = "stream";
+constexpr auto kStreamVersion = "v2";
+
+std::string GetSubscriptionMode(SubscribeRequest::SubscriptionMode mode) {
+  return mode == SubscribeRequest::SubscriptionMode::kSerial
+             ? kSubscriptionModeSerial
+             : kSubscriptionModeParallel;
+}
+
+std::string ToString(const boost::optional<std::string>& opt) {
```

Comment:
I meant the entire function is the same as boost::get_value_or("none"). There is not much sense to have a one-liner to save a couple of chars in the logging macro.